components/style/stylist.rs

```diff
@@ -98,6 +101,16 @@ pub struct Stylist {
 
     /// The total number of times the stylist has been rebuilt.
     num_rebuilds: usize,
+
+    /// The set of registered custom property associated with the document.
+    /// Should be initialized as soon as possible.
+    #[cfg_attr(feature = "servo", ignore_heap_size_of = "Arc")]
+    registered_property_set: Option<Arc<RwLock<RegisteredPropertySet>>>,
```

Comment:
Which SharedRwLock are you thinking of? Do you mean instead you'd just prefer if cascade did not do any locking at all wrt the RegisteredPropertySet? That would certainly make sense & is a good catch, thanks.